fix: LEAP-1363: View all is loading even when value is empty for RichText

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
When user is using a config with text, and uploading the data with no value for the text, the RichText tag is not loading and the prop isReady is not set to true, so when user clicks on `View all Annotations` view, it's not loading the task



#### What does this fix?
Just changes the validation from `!this_value` for `!this._value === null` so even if the value is empty but the data is correct and value exists, it's loading the tag and setting isReady to true when it's ready.

juliosgarbi committed:
fix: LEAP-1363: View all is loading even when value is empty for RichText

jombooth commented on web/libs/editor/src/tags/object/RichText/view.jsx:
```diff
@@ -362,7 +362,7 @@ class RichTextPieceView extends Component {
     const { item } = this.props;
 
-    if (!item._value) return null;
+    if (!item._value === null) return null;
```
would it be safer to go with `if (!item.value && item.value !== '')` to change the behavior a bit less?

hlomzik commented on web/libs/editor/src/tags/object/RichText/view.jsx:
```diff
@@ -362,7 +362,7 @@ class RichTextPieceView extends Component {
     const { item } = this.props;
 
-    if (!item._value) return null;
+    if (!item._value === null) return null;
```
this comparison doesn't make sense and will always return false because you are comparing boolean value with `null`. what are you trying to achieve here?

juliosgarbi replied:
my bad, the idea would be validate if `item._value` is null not `!item._value`. It's fixed

hlomzik replied:
if you are checking that value is missing (like it's `undefined` or `null`) then it's better to use `isDefined()`

juliosgarbi committed:
fix validation

juliosgarbi committed:
fix validation

juliosgarbi committed:
fix validation

juliosgarbi committed:
fix validation

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10392351373)
> Successfully merged:  delete mode 100644 web/libs/editor/src/common/Userpic/Userpic.styl

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-1363'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/10392351373

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10394484433)
> Successfully merged:  1 file changed, 4 insertions(+), 4 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-1363'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/10394484433

hlomzik commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10425827895)
> Successfully merged:  delete mode 100644 web/libs/editor/src/tags/object/Video/Video.styl

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1363'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/10425827895

Gondragos committed:
Fix error with empty spans removing

Gondragos committed:
Fix handling loaded state in rich text

we need to distinguish an empty string from non loaded state, for that we add ability to have null _value

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10470915990)
> Successfully merged:  create mode 100644 label_studio/ml_models/migrations/0010_modelinterface_skill_name.py

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1363'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/10470915990
